### PR TITLE
Move ScoredPointOffset into common

### PR DESCRIPTION
**Pull request by agourlay**
This PR moves `ScoredPointOffset` and its internal types into common.

The motivation is to enable the sparse vector search to return the same type as dense vector search.

### CODE REVIEW

File: lib/segment/src/vector_storage/vector_storage_base.rs [hunk 1]
```diff
@@ -1,40 +1,19 @@
...
 use super::quantized::quantized_vectors::QuantizedVectors;
 use super::simple_vector_storage::SimpleVectorStorage;
 use crate::common::Flusher;
 use crate::data_types::vectors::VectorElementType;
 use crate::entry::entry_point::OperationResult;
-use crate::types::{Distance, PointOffsetType, QuantizationConfig, ScoreType};
+use crate::types::{Distance, QuantizationConfig};
 use crate::vector_storage::appendable_mmap_vector_storage::AppendableMmapVectorStorage;
 
-#[derive(Copy, Clone, PartialEq, Debug, Default)]
```
Comment (agourlay) [inline lib/segment/src/vector_storage/vector_storage_base.rs:18]: the previous equivalent implementation for sparse vector can be deleted

File: lib/sparse/src/common/scored_candidate.rs [hunk 2]
```diff
@@ -1,25 +0,0 @@
-use std::cmp::Ordering;
-
-use ordered_float::OrderedFloat;
-
-use crate::common::types::RecordId;
-
-#[derive(Debug, PartialEq)]
-pub struct ScoredCandidate {
```
Comment (agourlay) [inline lib/sparse/src/common/scored_candidate.rs:8]: the previous equivalent implementation for sparse vector can be deleted

File: lib/sparse/src/index/posting_list.rs [hunk 3]
```diff
@@ -1,8 +1,10 @@
-use crate::common::types::{DimWeight, RecordId};
+use common::types::PointOffsetType;
+
+use crate::common::types::DimWeight;
 
 #[derive(Debug, Copy, Clone)]
 pub struct PostingElement {
-    pub record_id: RecordId,
```
Comment (agourlay) [inline lib/sparse/src/index/posting_list.rs:5]: the sparse vector index can now use the same type internally

**Comment by agourlay** [comment]
Setting myself up for a sweet rebase later in https://github.com/qdrant/qdrant/pull/2718

**Review by coszio — APPROVED** [review]
I like the initiative to begin moving some types to `common` crate :+1:.

As a provocative comment, why don't we set these migration up with one file per type / family of types? In this case we could start it by doing something like this:
```
common
└── src
    └── types
        ├── mod.rs
        ├── aliases.rs
        └── scored_point_offset.rs
```

This could help to write declaration with impls in separate files for each type.

**Comment by agourlay** [comment]
@coszio generally I prefer a pragmatic approach where we only migrate things when it is needed in small steps  :)

**Review by timvisee — APPROVED** [review]

**Pull request merged**